src/core/sys/posix/sys/statvfs.d

```diff
@@ -131,6 +131,66 @@ else version(NetBSD)
     int statvfs (const char * file, statvfs_t* buf);
     int fstatvfs (int fildes, statvfs_t *buf) @trusted;
 }
+else version (FreeBSD)
+{
+    enum  MFSNAMELEN = 16;
+    enum  MNAMELEN = 1024;
```

Comment:
`MNAMELEN` is defined as  `88` here: https://www.freebsd.org/cgi/man.cgi?query=statfs&sektion=2. Why are you changing it to `1024`?

Reply:
https://github.com/freebsd/freebsd/blob/aa552394dbf32a59a1ab8763f00c8fe774b22032/sys/sys/mount.h#L68

Reply:
I'm not sure why man page doesn't respect that, though :thinking:

Reply:
No he's right. 88 was the value in the 90's. The current value (used when the kernel is build) is well 1024: https://github.com/freebsd/freebsd/blob/master/sys/sys/mount.h#L68

Reply:
Ah, I think I know why. It might be the part of the inode64 changes for the FreeBSD. Here's the latest man page: https://github.com/freebsd/freebsd/blob/565b5a2e9ec6e684421a7e6eb7b37d19e3c0f47e/lib/libc/sys/statfs.2#L69

Reply:
(There's an issue about this, as right now dmd will not work on the CURRENT at all: https://issues.dlang.org/show_bug.cgi?id=17596)

Reply:
AFAIK, the way this usually works is by extracting the correct size for the from the host platform headers (considering there's no cross-compilation).

Reply:
I'm not sure how we should proceed here. Use only the latest definition, or provide the two under version FreeBSDv1X.X (or something like that). I would go for the second option, but I don't know if we have a good interface for passing this version to druntime build.

Reply:
The problem will arise as soon as this functionality is used in Phobos as the Autotester uses the (AFAIK) the older version of FreeBSD.

Reply:
Here's what @CyberShadow proposed in the issue:

```
- We add getosreldate and INO64_FIRST to druntime
- We add both the old and new struct definitions
- We add a stat wrapper which,when getosreldate() < INO64_FIRST, translates the old struct to the new struct and returns it.
```

Reply:
> There's an issue about this, as right now dmd will not work on the CURRENT at all: https://issues.dlang.org/show_bug.cgi?id=17596)

Thanks. @jercaianu can you add both definitions and add a D translation of the C wrapper function mentioned here: https://issues.dlang.org/show_bug.cgi?id=17596#c8 ?

Reply:
Sure. I'll look into it :)